storage/ndb/rest-server/data-access-rondb/src/db-operations/pk/common.cpp

```diff
@@ -364,78 +345,38 @@ RS_Status SetOperationPKCol(const NdbDictionary::Column *col, NdbOperation *oper
-      append the padding character "=" once or twice in the encoded string.
-
-      E.g. the string "a"
-        -> to byte array: [ 97 ]
-        -> to base64 string: "YQ=="
-      This uses "==" as padding, since we only have 1 byte to represent.
-
-      Since the function decoded_size() does not read the string, it is not aware
-      of whether it contains padding characters or not. PKValueLen() is also not
-      aware of this. Thereby, any "==" will be intepreted as part of the data
-      and therefore decoded_size() will always return a multiple of 3 bytes.
-
-      E.g. given base64 string "YQ==" (4 ASCII characters)
-        -> data_len == 4 bytes
-        -> decoded_size == 3 bytes (since base64 always uses 4 bytes to represent 3 bytes of data)
-    */
-    size_t decoded_size = boost::beast::detail::base64::decoded_size(data_len);
-
-    if (decoded_size > size_t(col->getLength())) {
-      
-      /*
-        As described above, it may be the case that the decoded_size falsely interpretes
-        1 or 2 padding bytes as part of the data. Since the decoded_size must be a
-        multiple of 3, it may now only be the next multiple after col->getLength().
-
-        E.g.: col->getLength() == 100; decoded_size is now only allowed to be 102
-      */
-      int base64Mod = col->getLength() % 3; // 100 % 3 == 1
-      int base64Total = col->getLength() - base64Mod + 3; // 100 - 1 + 3 == 102
-
-      if (decoded_size != size_t(base64Total)) {
-        return RS_CLIENT_ERROR(
-            std::string(ERROR_008) +
-            " Decoded data length (" + std::to_string(decoded_size) +
-            ") is greater than column length (" + std::to_string(col->getLength()) +
-            "). Column: " + std::string(col->getName())
-        );
-      }
-      /*
-        It may now still be the case that the real decoded_size is 1 byte larger than
-        the column length. But we cannot find out unless we decode the string.
-      */
-    }
-
-    // operation->equal expects a zero-padded char string
-    char pk[MAX_KEY_SIZE_IN_WORDS*4];
-    std::fill (pk, pk + col->getLength(), 0);
-
-    std::pair<std::size_t, std::size_t> ret = boost::beast::detail::base64::decode(pk, encodedStr, data_len);
-
-    // ret.first does not interpret padding bytes as part of data
-    if (ret.first > size_t(col->getLength())) {
-        return RS_CLIENT_ERROR(
-            std::string(ERROR_008) +
-            " ret.first (" + std::to_string(ret.first) +
-            ") is not equal to decoded_size (" + std::to_string(decoded_size)
-        );
+    /// Binary data is sent as base64 string
+    assert(col->getLength() <= BINARY_MAX_SIZE_IN_BYTES);
+    const char *encoded_str      = request->PKValueCStr(colIdx);
+    const size_t encoded_str_len = request->PKValueLen(colIdx);
+    size_t col_len               = col->getLength();
+
+    size_t decoded_size = boost::beast::detail::base64::decoded_size(encoded_str_len);
```

Comment:
I have an implementation of decode Base64 that we can easily convert such that it does both decode and checks that the length is ok. With this manner of doing it you will have to scan the binary array twice, one to calculate the length and one to decode.

So let's change to this version.